### ISSUE TITLE
Error in Media Library with limited permissions

### ISSUE THREAD

**Issue by ariadna-aldeguer** [strapi/strapi#20435]
## Bug report

- Node.js version: v18.20.2
- Strapi version: v4.21.1
- Database: MySql
- Is your project Javascript or Typescript: Typescript

### Describe the bug

With a role with permissions in plugins/upload of "access the media library" with any of these options
- can access when is creator
- can access when you have same role as creator
- both

Going to Media Library with a user with that role, it shows a pop up with "Warning, an error occurred", and it does not show files or folders on media library.

### Steps to reproduce the behavior

1. Go to Settings
2. Administration panel > Roles
3. Add new rol 
4. Plugins - Set only like the image 
![image](https://github.com/strapi/strapi/assets/71792438/686ce3ba-cdf3-4776-91d8-239c9542ff2a)
5. Click on settings in General (I have tried all, both, creator or same role)
![image](https://github.com/strapi/strapi/assets/71792438/f6dcea63-4e01-4954-850b-bf2fc519253c)
6. Access with user with this role
7. Go to Media Librery
8. Pop up with "Warning: An error ocurred"

If I create a folder and reload, the folder does not appear (As an administrator it is visible)

### Expected behavior

See folders and files creates by user or with the same role.

### Trace network

![image](https://github.com/strapi/strapi/assets/71792438/5b6d8cbf-b0ed-4862-a483-2853729b8167)

/admin/project-settings

```
{
    "data": null,
    "error": {
        "status": 403,
        "name": "ForbiddenError",
        "message": "Forbidden",
        "details": {}
    }
}
```

/upload/files?sort=createdAt:DESC&page=1&pageSize=10&filters[$and][0][folderPath][$eq]=/
```
{
    "data": null,
    "error": {
        "status": 500,
        "name": "InternalServerError",
        "message": "Internal Server Error"
    }
}
```

**Comment by Convly** [comment]
Thanks for reporting this issue

Closing as duplicate of https://github.com/strapi/strapi/issues/18297

A fix is on its way as per https://github.com/strapi/strapi/pull/20389

**Comment by ariadna-aldeguer** [comment]
> Thanks for reporting this issue
> 
> Closing as duplicate of #18297
> 
> A fix is on its way as per #20389

I have upgraded to version 4.25.1

If I assign 'Can access the media Library' when 'Has the same role as creator' or 'Is creater' the user can see all the folders, whether they are creators or not, either have the same role or not

![image](https://github.com/strapi/strapi/assets/71792438/9b848b19-eb2c-41a2-ab0a-b8e42af77f40)

![image](https://github.com/strapi/strapi/assets/71792438/0e587091-e822-48ee-adde-3bb7461162a5)

Furthermore, I can't find a way to see who created the folder or what role.
The following button has no response:

<img width="724" alt="image" src="https://github.com/strapi/strapi/assets/71792438/637295e1-389d-438f-802f-28863af7423d">
